cilium/cmd/bpf_ipmasq_list.go

```diff
@@ -26,7 +26,7 @@ var bpfIPMasqListCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		common.RequireRootPrivilege("cilium bpf ipmasq list")
 
-		cidrs, err := (&ipmasq.IPMasqBPFMap{}).Dump()
+		cidrs, err := (&ipmasq.IPMasqBPFMap{}).DumpForProtocols(getMasqueradingProtocols())
```

Comment:
`getMasqueradingProtocols` seems like a pretty big hammer to me. Could you instead call `IPMasq6Map` and `IPMasq4Map` (ignoring any "file doesn't exist" error) before `Dump()`?

Reply:
As discussed offline -- This won't work, because we're trying to `Open()` the map in `DumpWithCallback()`, so if we initialise `ipmasq.ipMasq6Map` by calling `ipmasq.IPMasq6Map()` in the CLI, we get an error:

```console
root@kind-worker:/home/cilium# cilium bpf ipmasq list
error dumping contents of map: loading pinned map /sys/fs/bpf/tc/globals/cilium_ipmasq_v6: no such file or directory
```

I'll follow your alternative suggestion instead, using the result of `ipv4Needed := ipmasq.IPMasq4Map().Open() == nil` (and v6) to check whether the maps are open and get the booleans to pass to `DumpForProtocols()`, instead of going for the API request.

Thanks for the review!